crates/weaver_forge/README.md

```diff
@@ -1,6 +1,217 @@
+  block_start: "{%"
+  block_end: "%}"
+  variable_start: "{{"
+  variable_end: "}}"
+  comment_start: "{#"
+  comment_end: "#}"
+
+# Please uncomment the following templates to override the default template
+# mapping. Each template mapping specifies a jaq filter (compatible with jq)
+# to apply to every file matching the pattern. The application_mode specifies
+# how the template should be applied. The application_mode can be `each` or
+# `single`. The `each` mode will evaluate the template for each object selected
+# by the jaq filter. The `single` mode will evaluate the template once with all
+# the objects selected by the jq filter.
+#
+# Note: jaq is a Rust reimplementation of jq. Most of the jq filters are
+# supported. For more information, see https://github.com/01mf02/jaq
+#
+# templates:
+#  - pattern: "**/registry.md"
+#    filter: "."
+#    application_mode: single
+#  - pattern: "**/attribute_group.md"
+#    filter: ".groups[] | select(.type == \"attribute_group\")"
+#    application_mode: each
+#  - pattern: "**/attribute_groups.md"
+#    filter: ".groups[] | select(.type == \"attribute_group\")"
+#    application_mode: single
+#  - pattern: "**/event.md"
+#    filter: ".groups[] | select(.type == \"event\")"
+#    application_mode: each
+#  - pattern: "**/events.md"
+#    filter: ".groups[] | select(.type == \"event\")"
+#    application_mode: single
+#  - pattern: "**/group.md"
+#    filter: ".groups[] | select(.type == \"group\")"
+#    application_mode: each
+#  - pattern: "**/groups.md"
+#    filter: ".groups[] | select(.type == \"group\")"
+#    application_mode: single
+#  - pattern: "**/metric.md"
+#    filter: ".groups[] | select(.type == \"metric\")"
+#    application_mode: each
+#  - pattern: "**/metrics.md"
+#    filter: ".groups[] | select(.type == \"metric\")"
+#    application_mode: single
+#  - pattern: "**/metric_group.md"
+#    filter: ".groups[] | select(.type == \"metric_group\")"
+#    application_mode: each
+#  - pattern: "**/metric_groups.md"
+#    filter: ".groups[] | select(.type == \"metric_group\")"
+#    application_mode: single
+#  - pattern: "**/resource.md"
+#    filter: ".groups[] | select(.type == \"resource\")"
+#    application_mode: each
+#  - pattern: "**/resources.md"
+#    filter: ".groups[] | select(.type == \"resource\")"
+#    application_mode: single
+#  - pattern: "**/scope.md"
+#    filter: ".groups[] | select(.type == \"scope\")"
+#    application_mode: each
+#  - pattern: "**/scopes.md"
+#    filter: ".groups[] | select(.type == \"scope\")"
+#    application_mode: single
+#  - pattern: "**/span.md"
+#    filter: ".groups[] | select(.type == \"span\")"
+#    application_mode: each
+#  - pattern: "**/spans.md"
+#    filter: ".groups[] | select(.type == \"span\")"
+#    application_mode: single
+```
+
+Supported case converters:
+- lowercase
+- UPPERCASE
+- PascalCase
+- camelCase
+- snake_case
+- SCREAMING_SNAKE_CASE
+- kebab-case
+- SCREAMING-KEBAB-CASE
+
+## Custom Filters
```

Comment:
Can you add:

- `split_ids`: Splits a string by '.' creating a list of nested ids.
- `flatten`:  Converts a List of Lists into a single list with all elements. e.g. `[[a,b],[c]] => [a,b,c]`


Also - I'm thinking we should drop "split_ids" for a more generic "split" filter that takes the pattern to split as argument....